### PR TITLE
[README] explain how to return different values for the same call

### PR DESCRIPTION
**Pull request by lyrixx**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -165,6 +165,13 @@ promise, there's plenty others you can use:
 Keep in mind, that you can always add even more promises by implementing
 `Prophecy\Promise\PromiseInterface`.
 
+If the method is called sereval times, you can use the following syntaxe to return different
```
Comment (odolbeau) [inline README.md]: syntax

File: README.md [hunk 2]
```diff
@@ -165,6 +165,13 @@ promise, there's plenty others you can use:
 Keep in mind, that you can always add even more promises by implementing
 `Prophecy\Promise\PromiseInterface`.
 
+If the method is called sereval times, you can use the following syntax to return different
```
Comment (Taluu) [inline README.md]:
```suggestion
If the method is called several times, you can use the following syntax to return different
```

File: README.md [hunk 3]
```diff
@@ -165,6 +165,13 @@ promise, there's plenty others you can use:
...
+values for each calls:
```
Comment (Pierstoval) [inline README.md]:
```suggestion
values for each call:
```

**Comment by stof** [comment]
I would rather document it as a note at the end of the `Method prophecies idempotency` section, as this feature is actually not recommended for most cases (relying on the order of calls for the same arguments tends to make test fragile, as adding one more call can break everything)

**Comment by lyrixx** [comment]
@stof You are right. I updated the PR

**Comment by gquemener** [comment]
Allow me to share this conversation where Konstantin gives more details about why this is, most of the time, not desirable to use such feature: https://gist.github.com/gquemener/292e7c5a4bbb72fd48a8

**Comment by lyrixx** [comment]
I often see people asking for that.  Could we merge this one since there is a warning?
Thanks

**Comment by lyrixx** [comment]
I often see people asking for that.  Could we merge this one since there is a warning?
Thanks

**Comment by lyrixx** [comment]
ping

File: README.md [hunk 4]
```diff
@@ -206,6 +206,17 @@ $user->setName('everzet')->will(function ($args) use ($user) {
...
+Note: If the method is called several times, you can use the following syntax to return different
+values for each call:
+
+```php
+$prophecy->read('123')->willReturn(1, 2, 3);
+```
+
+This feature is actually not recommended for most cases (relying on the order of
+calls for the same arguments tends to make test fragile, as adding one more call
+can break everything).
```
Comment (jakzal) [inline README.md]: I think the sentence in parenthesis should  just be a regular sentence. It's an important note.

Comment (lyrixx) [reply README.md]: @jakzal Thanks for your feedback. I removed parenthesis

**Pull request merged**